.github/workflows/ci.yml

```diff
@@ -87,6 +87,48 @@ jobs:
         if: steps.cache.outputs.cache-hit != 'true' && matrix.cache == 'true'
         run: rm -rf ~/.m2/repository/io/trino/trino-*
 
+  reproducible-build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # checkout all commits, as the build result depends on `git describe` equivalent
+          ref: |
+            ${{ github.event_name == 'repository_dispatch' &&
+                github.event.client_payload.pull_request.head.sha == github.event.client_payload.slash_command.args.named.sha &&
+                format('refs/pull/{0}/head', github.event.client_payload.pull_request.number) || '' }}
+      - uses: ./.github/actions/setup
+        with:
+          cache: 'false'
+          cleanup-node: true
```

Comment:
Redundant & takes time. Drop it

Reply:
Actually needed - there is no enough space to build twice

Reply:
Can we avoid pulling dependencies twice?
It would be faster if (a) we used the cache restore mode and (b) we used the downloaded dependencies in both build steps. Either by configuring maven to use ~/.m2/repository as a repo, or by hardlinking/symlinking from ~/.m2/repository to separate build directories.